### PR TITLE
chore: Release 0.15.0-0 🚀

### PR DESCRIPTION
**Pull request by sumitramanga**
### Description :memo:
- **Purpose**: We are releasing version `0.15.0-0` of the provider including resolving #146
- **Approach**: Merge all changes in `develop` into `master`

**Type of change**
- [x] Release

**Release contents**
:point_right: Add support to async/await to `send` API method
:point_right: Fix avoiding infinite recursion in` filterKeys` method
:point_right: Various documentation and workflow changes
:point_right: Dev dependency updates 

### Test plan :test_tube:
- Run example apps and ensure that test exceptions are sending to Raygun
- Run the provider on clean project locally
- Use the `send` method in the example apps and clean project

### Author to check :eyeglasses:
- [ ] Project and all contained modules builds successfully
- [ ] Self-/dev-tested 
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [ ] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

**Comment by sumitramanga** [comment]
Hi @paul-uz, we've managed to push foward with getting a release ready for your requested Issue  https://github.com/MindscapeHQ/raygun4node/pull/181

We should have this out soon!

**Review by miquelbeltran — CHANGES_REQUESTED** [review]
Something seems wrong about this PR, merging `develop` into `master` again shouldn't show all these changes. I think this is due to the rebasing of `master`. Probably this can be fixed by merging `master` into `develop` first with a merge commit, otherwise commit history has diverged.

Secondly, the release process should be done in a two-step process:

1st. We need to increase the version in the `package.json` and update the CHANGELOG, on a separated PR into `develop`. This is necessary so the version change is found on both branches.

2nd. We do the merge of `develop` into `master` effectively doing the release merge.

This is something that git-flow would solve for us as well.

I'd recommend closing this PR and follow those steps.

**Comment by miquelbeltran** [comment]
The version preparing PR is here: https://github.com/MindscapeHQ/raygun4node/pull/202

**Comment by miquelbeltran** [comment]
We can keep this PR open, I will fix the diverged history between master and develop in a new PR meanwhile.

**Comment by miquelbeltran** [comment]
And the PR fixing the merge conflicts: https://github.com/MindscapeHQ/raygun4node/pull/203

**Comment by paul-uz** [comment]
I'm confused, is the latest version 0.14.0 or 0.15.0?

**Comment by miquelbeltran** [comment]
> I'm confused, is the latest version 0.14.0 or 0.15.0?

0.14.0 has been just released, and we are preparing 0.15.0-0 with the async/await change

**Comment by miquelbeltran** [comment]
well, the PR I did actually doesn't fix the merge conflicts

**Review by miquelbeltran — APPROVED** [review]
LGTM, tested locally with both examples

**Comment by miquelbeltran** [comment]
remember, we need to do a "Merge pull request" ;)

**Comment by sumitramanga** [comment]
@miquelbeltran Will do, thank you for the reminder!

**Review by PanosNB — CHANGES_REQUESTED** [review]
All good, great work! One things to consider please around cycle prevention and not deleting ancestor references from the set. We could also add the case of two or more backwards references to the test (great to see that there!)

**Review by sumitramanga — COMMENTED** [review]
I can't approve my own PR but this looks good to me! Panos raises a good point so that's something to look into

File: lib/raygun.messageBuilder.ts [hunk 1]
```diff
@@ -31,18 +31,37 @@ type UserMessageData = RawUserData | string | undefined;
...
-  Object.keys(obj).forEach(function (i) {
-    if (filters.indexOf(i) > -1) {
-      delete _obj[i];
+
+  Object.keys(obj).forEach(function (key) {
+    // Remove child if:
+    // - the key is in the filter array
+    // - the value is already in the explored Set
+    if (filters.indexOf(key) > -1 || _explored.has(_obj[key])) {
+      delete _obj[key];
```
[the remark this answers is not in the document]
Comment (PanosNB) [reply lib/raygun.messageBuilder.ts:62]: My bad, looking back into this, I see now that we do not delete the _explored set. All good, I will approve

**Review by PanosNB — APPROVED** [review]
All good, I had misunderstood smth before. Great work!

**Review by TheRealAgentK — APPROVED** [review]
Ok, LGTM! Let's release it.

**Comment by paul-uz** [comment]
Can I ask why this is marked as `0.15.0-0` and not just `0.15.0`? It means I can no longer use `^0.15.0` in my package file

**Comment by miquelbeltran** [comment]
Hi @paul-uz, we have released the async/await send method support as pre-release (that's why the `-0` at the end) to collect feedback from users. Once we are sure it works as expected, we will do a proper stable release.

In the near future, the Raygun team plans to do a proper `1.0.0` eventually, nevertheless.

**Pull request merged**